src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/storage/S3StorageService.kt

```diff
@@ -18,30 +18,18 @@ class S3StorageService(
 
   private val log = LoggerFactory.getLogger(this::class.java)
 
-  override fun downloadFile(fileKey: UUID): ByteArray {
-    return s3Client.getObject(
-      GetObjectRequest.builder().bucket(bucketName).key(fileKey.toString()).build()
-    ).readAllBytes()
+  override fun downloadFile(fileS3Key: UUID): ByteArray {
+    log.debug("downloading file to s3://$bucketName/$fileS3Key")
+    val request = GetObjectRequest.builder().bucket(bucketName).key(fileS3Key.toString()).build()
+    return s3Client.getObject(request).readAllBytes()
   }
 
-  override fun uploadFile(fileBytes: ByteArray): S3BulkResponseEntity {
-    log.debug("Uploading file to s3://$bucketName")
-    val result: S3BulkResponseEntity
-    val uuid = UUID.randomUUID()
-    s3Client.putObject(
-      PutObjectRequest.builder().bucket(bucketName).key(uuid.toString()).build(),
-      RequestBody.fromBytes(fileBytes)
-    )
-      .sdkHttpResponse()
-      .also { log.info("uploaded $uuid to $bucketName: ${it.statusCode()}") }
-      .let { response ->
-        result = S3BulkResponseEntity(
-          bucket = bucketName,
-          fileKey = uuid,
-          successful = response.isSuccessful,
-          statusCode = response.statusCode()
-        )
-      }
-    return result
+  override fun uploadFile(fileBytes: ByteArray): UUID {
+    val fileS3Key = UUID.randomUUID()
+    log.debug("Uploading file to s3://$bucketName/$fileS3Key")
+    val request = PutObjectRequest.builder().bucket(bucketName).key(fileS3Key.toString()).build()
+    val body = RequestBody.fromBytes(fileBytes)
+    s3Client.putObject(request, body)
```

Comment:
is there anything that is returned by putObject that we should be checking?

Reply:
I had a look at the doc: `putObject()` throws different AWS specific exceptions in case of errors - so we don't need to check the internal HTTP status codes to confirm that the operation was successful ; we can just handle the exceptions. Perhaps we should encapsulate these exceptions with a specific exception like `UploadError`?

Reply:
From the Javadoc:
```
Throws:
SdkClientException - If any errors are encountered in the client while making the request or handling the response.
AmazonServiceException - If any errors occurred in Amazon S3 while processing the request.
```